benchmarks/src/main/scala/akka.kafka.benchmarks/KafkaConsumerBenchmarks.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka.benchmarks
+
+import java.util
+import java.util.concurrent.Semaphore
+
+import com.codahale.metrics.Meter
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.kafka.clients.consumer.{OffsetCommitCallback, OffsetAndMetadata}
+import org.apache.kafka.common.TopicPartition
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+object KafkaConsumerBenchmarks extends LazyLogging {
+  val pollTimeoutMs = 50L
+
+  /**
+   * Reads messages from topic in a loop, then discards immediately. Does not commit.
+   */
+  def consumePlainNoKafka(fixture: KafkaConsumerTestFixture, meter: Meter): Unit = {
+
+    @tailrec
+    def pollInLoop(readLimit: Int, readSoFar: Int = 0): Int = {
+      if (readSoFar >= readLimit)
+        readSoFar
+      else {
+        logger.debug(s"Polling")
+        meter.mark()
+        pollInLoop(readLimit, readSoFar + 1)
+      }
+    }
+    meter.mark()
+    pollInLoop(readLimit = fixture.msgCount)
+  }
+
+  /**
+   * Reads messages from topic in a loop, then discards immediately. Does not commit.
+   */
+  def consumePlain(fixture: KafkaConsumerTestFixture, meter: Meter): Unit = {
+    val consumer = fixture.consumer
+
+    @tailrec
+    def pollInLoop(readLimit: Int, readSoFar: Int = 0): Int = {
+      if (readSoFar >= readLimit)
+        readSoFar
+      else {
+        logger.debug(s"Polling")
+        val records = consumer.poll(pollTimeoutMs)
+        val recordCount = records.count()
+        records.iterator().toList // ensure records are processed
+        meter.mark(recordCount.toLong)
+        logger.debug(s"${readSoFar + recordCount} records read. Limit = $readLimit")
+        pollInLoop(readLimit, readSoFar + recordCount)
+      }
+    }
+    meter.mark()
+    pollInLoop(readLimit = fixture.msgCount)
+    fixture.close()
+  }
+
+  /**
+   * Reads messages from topic in a loop and groups in batches of given max size. Once a batch is completed,
+   * batches the next part.
+   */
+  def consumerAtLeastOnceBatched(batchSize: Int)(fixture: KafkaConsumerTestFixture, meter: Meter): Unit = {
+    val consumer = fixture.consumer
+
+    var lastProcessedOffset = 0L
+    var accumulatedMsgCount = 0L
+    val semaphore = new Semaphore(1)
+
+    def doCommit(): Unit = {
+      semaphore.acquire()
+      accumulatedMsgCount = 0
+      val offsetMap = Map(new TopicPartition(fixture.topic, 0) -> new OffsetAndMetadata(lastProcessedOffset))
+      logger.debug("Committing offset " + offsetMap.head._2.offset())
```

Comment:
Yes, macros should do the work. Does `ActorLogging` require `if (logger.isDebugEnabled)`?